### PR TITLE
Replaced get_samples in CRUD to work with Illumina

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
Replaced old CRUD function `get_samples_from_flow_cell` with `get_samples_by_illumina_flow_cell_internal_id` which works with the new database models.

### Changed

- Updated CLI commands with the new CRUD function:
  - `cg backup retrieve-spring-files`
  - `cg decompress flow-cell`
  - `cg store flow-cell`
  - `cg store demultiplexed-flow-cell`
- Updated tests 

### Fixed

- Remove idling method `_get_samples` defined in the CLI module and redefined it in the CRUD as `get_samples_by_identifier`
- Removed old store fixtures

## Review

- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- Merging to development branch, no deployment yet

### CODE REVIEW

File: cg/store/crud/read.py [hunk 1]
```diff
@@ -471,11 +471,15 @@ def get_flow_cells_by_case(self, case: Case) -> list[Flowcell] | None:
...
-            return flow_cell.samples
+    def get_samples_by_illumina_flow_cell_internal_id(
+        self, flow_cell_id: str
+    ) -> list[Sample] | None:
+        """Return samples present on an Illumina flow cell."""
+        sequencing_run: IlluminaSequencingRun = (
+            self.get_illumina_sequencing_run_by_device_internal_id(device_internal_id=flow_cell_id)
+        )
+        if sequencing_run:
+            return [metric.sample for metric in sequencing_run.sample_metrics]
```
[the remark this answers is not in the document]
Comment (diitaz93) [reply cg/store/crud/read.py:482]: True, it looks much better. I will change it

File: cg/store/crud/read.py [hunk 2]
```diff
@@ -471,11 +471,15 @@ def get_flow_cells_by_case(self, case: Case) -> list[Flowcell] | None:
...
         ).all()
 
-    def get_samples_from_flow_cell(self, flow_cell_id: str) -> list[Sample] | None:
-        """Return samples present on flow cell."""
-        flow_cell: Flowcell = self.get_flow_cell_by_name(flow_cell_name=flow_cell_id)
-        if flow_cell:
-            return flow_cell.samples
+    def get_samples_by_illumina_flow_cell_internal_id(
+        self, flow_cell_id: str
+    ) -> list[Sample] | None:
```
Comment (Vince-janv) [inline cg/store/crud/read.py:476]: Isn't the "by" part clear by what you pass as to the function?
```suggestion
    def get_samples_by_illumina_flow_cell(
        self, flow_cell_interna_id: str
    ) -> list[Sample] | None:
```

**Review by Vince-janv — APPROVED** [review]
👍

File: cg/store/crud/read.py (hunk 2 above)
Comment (diitaz93) [reply cg/store/crud/read.py:476]: I will include your suggestion in the following PR

File: cg/store/crud/read.py [hunk 3]
```diff
@@ -1059,6 +1063,16 @@ def get_sample_by_internal_id(self, internal_id: str) -> Sample | None:
...
 
+    def get_samples_by_identifier(self, object_type: str, identifier: str) -> list[Sample]:
+        """Return all samples from a flow cell, case or sample id"""
+        object_to_filter: dict[str, Callable] = {
+            "sample": self.get_sample_by_internal_id,
+            "case": self.get_samples_by_case_id,
+            "flow_cell": self.get_samples_by_illumina_flow_cell_internal_id,
+        }
+        samples: Sample | list[Sample] = object_to_filter[object_type](identifier)
+        return samples if isinstance(samples, list) else [samples]
```
[the remark this answers is not in the document]
Comment (diitaz93) [reply cg/store/crud/read.py:1074]: in `cg/cli/backup.py:retrieve_spring_files`

**Pull request merged**